src/components/application_manager/src/commands/command_impl.cc

```diff
@@ -130,6 +175,52 @@ void CommandImpl::ReplaceMobileByHMIAppId(
   }
 }
 
+bool CommandImpl::ReplaceHMIWithMobileAppId(
+    NsSmartDeviceLink::NsSmartObjects::SmartObject& message) {
+  if (message.keyExists(strings::app_id)) {
+    ApplicationSharedPtr application =
+        application_manager_.application_by_hmi_app(
+            message[strings::app_id].asUInt());
+
+    if (!application) {
+      LOG4CXX_ERROR(logger_, "Substitution HMI --> mobile id is failed.");
+      return false;
+    }
+    LOG4CXX_DEBUG(logger_,
+                  "ReplaceHMIWithMobileAppId from "
+                      << message[strings::app_id].asInt() << " to "
+                      << application->app_id());
+    message[strings::app_id] = application->app_id();
+  } else {
+    switch (message.getType()) {
+      case smart_objects::SmartType::SmartType_Array: {
+        smart_objects::SmartArray* message_array = message.asArray();
+        smart_objects::SmartArray::iterator it = message_array->begin();
+        for (; it != message_array->end(); ++it) {
+          if (!ReplaceHMIWithMobileAppId(*it)) {
+            break;
+          }
+        }
+        break;
+      }
+      case smart_objects::SmartType::SmartType_Map: {
+        std::set<std::string> keys = message.enumerate();
+        std::set<std::string>::const_iterator key = keys.begin();
+        for (; key != keys.end(); ++key) {
+          if (!ReplaceHMIWithMobileAppId(message[*key])) {
+            break;
+          }
+        }
+        break;
+      }
+      default: { break; }
+    }
+  }
+
+  return true;
+}
+
+// DEPRECATED
 void CommandImpl::ReplaceHMIByMobileAppId(
```

Comment:
Same comment as with `ReplaceMobileByHMIAppId`, this can be turned into a wrapper method for the new function.

Reply:
@jacobkeeler For this method we can use wrapper with no problem. But for similar method `ReplaceMobileByHMIAppId` we can not. And therefore in case we change it - as result there will be mix of wrapped & unwrapped methods which will reduce readability and consistency. Moreover all `DEPRECATED` methods will be deleted in future.

Reply:
I personally think that the cleaner option would be to use wrapped methods where possible for deprecated methods. There are very few functions where this cannot be done, and as I mention in the previous comment, there is no reason that this cannot be done for `ReplaceMobileByHMIAppId`.

Reply:
@jacobkeeler In case we create wrappers for deprecated methods - as result we have clean deprecated methods which can be safely deleted and some deprecated wrappers which will contain needed , not deprecated code . And when the time to delete will come - we have to check and analyze all deprecated methods in our code if they contain needed(not deprecated code) and extract it to separate functions , as well as change usage again to not deprecated ones. This is real overhead. 
Especially if someone from other team will be asked to remove deprecated methods.

Reply:
@jacobkeeler Fixed in 88fc4e0d5a15d60bd2edfc9bd623a54bd71d0243